### PR TITLE
Homework[Shapes] - Modern C++

### PR DESCRIPTION
**Pull request by Pawellos**

### CODE REVIEW

File: shapes/main.cpp [hunk 1]
```diff
@@ -1,79 +1,88 @@
...
+// Task 11 - Sfinae
+template <typename T, typename = typename std::enable_if<std::is_base_of<Shape, T>::value>>
+void insertOnlyShapes(Collection& collection, std::shared_ptr<T> shape) {
+    collection.emplace_back(shape);
+}
+
+// Task 18 - Variadic template
+template <class DerivedType, class... Arguments>
+std::shared_ptr<Shape> make_shape(Arguments&&... args) {
+    return make_shared<DerivedType>(decltype(args)(args)...);
```
Comment (ziobron) [inline shapes/main.cpp]: the best way is to use perfect forwarding with std::forward. Without it you always pass l-value references.
```suggestion
    return make_shared<DerivedType>(std::forward<Arguments>(args)...);
```

**Review by ziobron — CHANGES_REQUESTED** [review]
one small thing to correct

File: shapes/main.cpp (hunk 1 above)
Comment (Pawellos) [reply shapes/main.cpp]: Thank you for letting me know, It is still magic for me :)

**Review by MaciejSalamonski — APPROVED** [review]
:+1: @Pawellos

**Review by WBigaj — COMMENTED** [review]
I have checked your code and have nothing to add that was not mentioned by previous reviewers.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
15 XP za Nowoczesny C++ SFINAE (4447)
5 XP za Nowoczesny C++ alignas (4448)
5 XP za Nowoczesny C++ delegating constructors (4449)
5 XP za Nowoczesny C++ lambda (4450)
5 XP za Nowoczesny C++ structured bindings (4451)
15 XP za Nowoczesny C++ variadic templates (4452)

{ "requirement_ids": [4447, 4448, 4449, 4450, 4451, 4452] }

**Pull request closed without merging**